addons/t9n/static/src/core/project_list.js

```diff
@@ -0,0 +1,87 @@
+import { Component, useState, onMounted } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+import { formatList } from "@web/core/l10n/utils";
+
+export class ProjectList extends Component {
+    static template = "t9n.ProjectList";
+
+    setup() {
+        this.action = useService("action");
+        this.state = useState({
+            projects: [],
+            filters: {
+                searchText: "",
+            },
+            sorting: {
+                column: "name",
+                order: "asc",
+            },
+        });
+        this.store = useState(useService("t9n.store"));
+        onMounted(async () => {
+            await this.fetchProjects();
+        });
+    }
+
+    async fetchProjects() {
+        const searchDomain = [];
+        if (this.state.filters.searchText) {
+            searchDomain.push(["name", "ilike", this.state.filters.searchText]);
+        }
+        const projects = await this.store.fetchProjects(searchDomain);
+        projects.sort((p1, p2) => {
+            let p1Col = p1[this.state.sorting.column];
+            let p2Col = p2[this.state.sorting.column];
+
+            if (this.state.sorting.column !== "resource_count") {
+                p1Col = p1Col.toLowerCase();
+                p2Col = p2Col.toLowerCase();
+            }
+
+            if (p1Col < p2Col) {
+                return this.state.sorting.order === "asc" ? -1 : 1;
+            }
+            if (p1Col > p2Col) {
+                return this.state.sorting.order === "asc" ? 1 : -1;
+            }
+            return 0;
+        });
+        this.state.projects = projects;
+        this.formatTargetLanguages();
+    }
+
+    formatTargetLanguages() {
+        for (const project of this.state.projects) {
+            project.target_langs = formatList(
+                project.target_langs.map((lang) => {
+                    return lang.name;
+                })
+            );
+        }
+    }
+
+    projectClicked(id) {
+        this.action.doAction({
+            type: "ir.actions.act_window",
+            res_id: id,
+            res_model: "t9n.project",
+            views: [[false, "form"]],
+            target: "new",
+        });
+    }
+
+    async handleSearchInput(event) {
+        this.state.filters.searchText = event.target.value.trim();
```

Comment:
You could put `t-model.trim="searchText"` directly in the template on the input instead